### ISSUE TITLE
template-function is an unknown keyword with `@version: 3.6`

### ISSUE THREAD

**Issue by vincentbernat** [balabit/syslog-ng#796]
When upgrading from 3.6 to 3.7, I get the following error:

```
Error parsing config, root plugin template-function not found in /home/bernat/.virtualenvs/syslog-ng/share/include/scl/cim/template.conf at line 23, column 1:
                                                   included from /home/bernat/.virtualenvs/syslog-ng/etc/scl.conf line 32, column 1
                                                   included from /home/bernat/Documents/exoscale/S03-2015/syslog-ng.conf line 4, column 1

template-function "format-cim" "$(format-json --pair @timestamp='${R_ISODATE}' --pair @message='${MSG}' --key .cim.* --shift 5 --key _* --key .* --replace-prefix .=_ --key *.*)\n";
^^^^^^^^^^^^^^^^^
```

The `template-function` keyword is unknown when using `@version: 3.6`. I didn't find a way to make this part of the configuration active only when `@version` is 3.7 or more recent.

**Comment by faxm0dem** [comment]
This one has become a classic at $WORK
:+1:

**Comment by bazsi** [comment]
Hi,

I've improved my original version (from here:
https://lists.balabit.hu/pipermail/syslog-ng/2015-March/021932.html), which
you can find on this branch:

https://github.com/balabit/syslog-ng/tree/f/allow-new-keywords-with-old-config-versions

Since it's a largish change I would love to see actual, real-world configs
be tested with this patch applied. a "syslog-ng -s" to check syntax should
suffice. Of course careful code review might also prove it is correct.

Feedback appreciated.
@lbudai, @ihrwein


-- 
Bazsi

**Comment by ihrwein** [comment]
Bazsi's mentioned branch has been merged. @vincentbernat could you close the issue, please?

**Comment by vincentbernat** [comment]
Ok!

**Comment by phemmer** [comment]
What version is the fix included in? The linked branch no longer exists, so I cant search for the code to see if it's present.
I'm running syslog-ng 3.7.3 and still seeing this issue.

**Comment by MrAnno** [comment]
Hello @phemmer,

PR #828 should be the one you are looking for. It's not included in 3.7.3.
